src/algebraic_geometry/locally_ringed_space/has_colimits.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import algebraic_geometry.locally_ringed_space
+import algebra.category.CommRing.constructions
+import algebraic_geometry.open_immersion
+import category_theory.limits.constructions.limits_of_products_and_equalizers
+
+/-!
+# Colimits of LocallyRingedSpace
+
+We construct the explict coproducts and coequalizers in the category of `LocallyRingedSpace`.
+It then follows that `LocallyRingedSpace` has all colimits, and
+`forget_to_SheafedSpace` preserves them.
+-/
+
+namespace algebraic_geometry
+
+universes v u
+
+open category_theory category_theory.limits opposite
+
+namespace SheafedSpace
+
+variables {C : Type u} [category.{v} C] [has_limits C]
+variables {ι : Type v} (F : discrete ι ⥤ SheafedSpace C)
+
+lemma sigma_ι_jointly_surjective (x : colimit F) :
+  ∃ (i : ι) (y : F.obj i), (colimit.ι F i).base y = x :=
+begin
+  let e : (colimit F).carrier ≅ _ :=
+    preserves_colimit_iso (SheafedSpace.forget _) _ ≪≫
+      colim.map_iso discrete.nat_iso_functor ≪≫ Top.sigma_iso_sigma _,
+  refine ⟨(e.hom x).1, (e.hom x).2, _⟩,
+  apply (Top.mono_iff_injective e.hom).mp infer_instance,
+  generalize : e.hom x = y,
+  rw ← Top.comp_app,
+  erw [ι_preserves_colimits_iso_hom_assoc (SheafedSpace.forget _), colimit.ι_map_assoc,
+    Top.sigma_iso_sigma_hom_ι_apply],
+  cases y,
+  refl
+end
+
+instance {X Y : SheafedSpace C} (f g : X ⟶ Y) : epi (coequalizer.π f g).base :=
+begin
+  erw ← (show _ = (coequalizer.π f g).base, from
+    ι_comp_coequalizer_comparison f g (SheafedSpace.forget C)),
+  rw ← preserves_coequalizer.iso_hom,
+  apply epi_comp
+end
+
+end SheafedSpace
+
+namespace LocallyRingedSpace
+
+section has_coproducts
+
+variables {ι : Type u} (F : discrete ι ⥤ LocallyRingedSpace.{u})
+
+/-- The explicit coproduct for `F : discrete ι ⥤ LocallyRingedSpace`. -/
+noncomputable
+def coproduct : LocallyRingedSpace :=
+{ to_SheafedSpace := colimit (F ⋙ forget_to_SheafedSpace : _),
+  local_ring := λ x, begin
+    rcases SheafedSpace.sigma_ι_jointly_surjective (F ⋙ forget_to_SheafedSpace) x with ⟨i, y, ⟨⟩⟩,
```

Comment:
I think using `obtain ... := ...` looks nicer, but `rcases` is perfectly fine if you prefer to keep it like this.